### PR TITLE
feat(halyard): Add support for deprecating flags

### PR DESCRIPTION
**Pull request by sbwsg**
Twin of https://github.com/spinnaker/deck/pull/5894

Appengine Container URL deploys will now always be available in deck.

### CODE REVIEW

**Comment by sbwsg** [comment]
Does Hal have a flag deprecation process?  If a consumer is using this feature flag in a script then I suppose that script will now be broken.

**Comment by lwander** [comment]
Not yet for flags, but yet for commands: https://github.com/spinnaker/halyard/blob/master/docs/deprecating-commands.md

What I would do is look at how @danielpeach implemented the ValidForSpinnakerVersion annotation, and add "upperBound" support: https://github.com/spinnaker/halyard/blob/master/halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java#L62

**Comment by sbwsg** [comment]
@lwander @danielpeach I've extended the ValidForSpinnakerVersion annotation to add a `upperBound` and `tooHighMessage`.

It manifests as follows:

screen shot 2018-10-25 at 3 48 01 pm

PTAL

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java [hunk 1]
```diff
@@ -47,19 +47,19 @@
   String context;
   String cluster;
   String user;
-  @ValidForSpinnakerVersion(lowerBound = "1.5.0", message = "Spinnaker does not support configuring this behavior before that version.")
+  @ValidForSpinnakerVersion(lowerBound = "1.5.0", upperBound = "", tooLowMessage = "Spinnaker does not support configuring this behavior before that version.", tooHighMessage = "")
```
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java]: I had assumed there was a way to keep parameters default/non-required -- do we need this `upperBound = ""` for each use of the annotation?

Comment (sbwsg) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java]: I likewise assumed I could declare some `default` methods (ala `getUpperBound()`, `setUpperBound(String)`) on the interface but wasn't able to get the syntax into a format that intellij accepted.

Comment (sbwsg) [reply halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/providers/kubernetes/KubernetesAccount.java]: Ach, got it, @danielpeach pointed me towards how to declare defaults in annotation.

**Pull request merged**